pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -118,6 +118,10 @@ func (r *pullthroughBlobStore) proxyStat(ctx context.Context, retriever importer
 }
 
 // ServeBlob attempts to serve the requested digest onto w, using a remote proxy store if necessary.
+// Important! This function is called for GET and HEAD requests. Docker client uses[1] HEAD request
```

Comment:
@mfojtik Now the mirroring process starts when you push the image (docker uses HEAD to check layers). So the base image will begin to download immediately. I don't think it's wrong because once you push the image you want to pull it after a while.